src/EnergyPlus/Pumps.hh

```diff
@@ -91,16 +91,20 @@ namespace Pumps {
     };
 
     extern std::string const cPump_VarSpeed;
-    extern int const Pump_VarSpeed;
     extern std::string const cPump_ConSpeed;
-    extern int const Pump_ConSpeed;
     extern std::string const cPump_Cond;
-    extern int const Pump_Cond;
     extern std::string const cPumpBank_VarSpeed;
-    extern int const PumpBank_VarSpeed;
     extern std::string const cPumpBank_ConSpeed;
-    extern int const PumpBank_ConSpeed;
-    extern Array1D_string const cPumpTypes;
+
+    enum class iPumpType : int
+    {
+        Unassigned = 0,
+        VarSpeed = 101,
+        ConSpeed = 102,
+        Cond = 103,
+        Bank_VarSpeed = 104,
+        Bank_ConSpeed = 105
+    };
```

Comment:
Try to make these just 1-5, instead of 101-105. See what happens. Live a little!

Reply:
Will remember consistency is key!

Reply:
I'd say even go farther and do -1 for the invalid case and 0-4 for the valid entries.

Reply:
Just beware any if statements that may be checking for `=0` or `<0`  or `<100` etc. instead of testing against an enum name.

Reply:
Thanks. Most of these comparison checks usually pop up as IDE/compiler errors when we convert the `extern int const` definitions to `enum`s, saying that enums cannot be compared with ints. I have tried to refactor any checks like `<0` and `<100` to check against the actual enumerators, but I will keep an eye out. If I did things right (fingers crossed), converting the (101-105) to (-1 to 4) should be simple and I would not have to make changes anywhere else, since the associated `int`s are not being used for comparisons anymore. Thank you for checking and the advice,

Reply:
What is the i in `iPumpType`? `int`? I'm not against [Hungarian Notation](https://en.wikipedia.org/wiki/Hungarian_notation) (they made us use it at Microsoft back in the day) but I didn't think we use it in EnergyPlus. Neither here nor there.

Reply:
That's probably a precedent I started or at least perpetuated. I believe I kept running into naming conflicts, so I just added the "i" to help distinguish the enum.

Reply:
Let's please not use Hungarian Notation. There are other ways to address conflicts.

Reply:
I think another small advantage being the distinction between enum class names and variables that use them (like [here](https://github.com/NREL/EnergyPlus/blob/f4a11a4781cf31579144724e559705ed1b9ad4f2/src/EnergyPlus/Pumps.hh#L148)). If there is a standard format for naming enum class names that we would like to follow, I'm glad to change to that.

Reply:
So, now that I think of it this isn't really HN, because `iPumpType` is a type not a variable. If this were HN, it would be `PumpType_t`. @jasondegraw, your thoughts?

Reply:
I'm pretty sure that a more standard approach would be to use lower case initial characters for variables (`thisIsMyVariable`) and upper case initial characters for types (`ThisIsMyType`).

Reply:
I support that simple style.  In this case we are already in the pump's namespace, so you could probably go pretty simple and still have clarity:

```
namespace EnergyPlus {
 namespace Pumps {
  enum class Type { Constant, Variable, Etc }; // or PumpType if you really want
  struct PumpData {
   ...
   Type type; // or myType if you want
   ...
  }
 }
}

Reply:
I'll name them to whatever convention is generally agreed upon as best practice. I'm not aware of anything in our Wiki that describes this, although, those pages are somewhat out of date. I believe I was just looking at the Google C++ style guide on enumerators (https://google.github.io/styleguide/cppguide.html#Enumerator_Names) that says they can be named like constants (see here: https://google.github.io/styleguide/cppguide.html#Constant_Names)

Reply:
@amirroth I'd take `Type_t` over `iType` any day of the week.

Reply:
@mitchute I don't really find that to be too persuasive. I believe I've seen cases where the enumeration is getting named something like `iThingTypeNum`. It's no longer a number, so the obvious name to me would be `Thing` or `ThingType`. I've been slowly changing variable names to start with lower case as I work on objects and starting out that way with new objects, so from my point of view this method of naming is heading the wrong way.

Reply:
You'll get no disagreement from me that there are plenty of bad examples of type names inside of EnergyPlus. In fact, that's probably what lead me down this road to begin with. Probably something like ```PumpType``` not being an enum type, but being a poorly named struct or local variable that was causing naming problems. This was a quick way of getting around the issue. Arguably it wasn't the best, but it was *A* way.

That being said, I don't think the leading "i" or "k" or whatever is really that bad. At least for me, since I've been working with these so much over the last year, my brain is already programmed to understand that the leading "i" means that I'm dealing with an enumerator, and that I don't have to drill in and see whether this really is an enumerator "type", or some other poorly named struct.

However, as I said, I'll do whatever.

Reply:
@Myoldmopar, so with your example "PumpType" enumerations would be `Type::Constant` within the Pumps namespace and `Pumps::Type::Constant` outside of it? I don't know if I like that... 

PS I would similarly complain about the type name itself but I realize that all of that is just going to be `auto` anyway so what's the point ...